backend/protocol_rpc/fastapi_server.py

```
@@ -0,0 +1,114 @@
+    allow_credentials=True,
+    allow_methods=["*"],
+    allow_headers=["*"],
+)
+
+# Include health check endpoints
+app.include_router(health_router)
+
+
+@app.get("/ready")
+async def readiness_check_with_app_state(
+    rpc_router: FastAPIRPCRouter | None = Depends(get_rpc_router_optional),
```

Ruff (0.13.1)

Do not perform function call `Depends` in argument defaults; instead, perform the call within the function, or read the default from a module-level singleton variable

(B008)